server/service/queries.go

```diff
@@ -287,10 +291,25 @@ func (svc *Service) ModifyQuery(ctx context.Context, id uint, p fleet.QueryPaylo
 	}
 	if p.Logging != nil {
 		query.Logging = *p.Logging
+		if *p.Logging != fleet.LoggingSnapshot {
+			shouldDiscardQueryResults = true
+		}
 	}
 	if p.ObserverCanRun != nil {
 		query.ObserverCanRun = *p.ObserverCanRun
 	}
+	if p.DiscardData != nil {
+		query.DiscardData = *p.DiscardData
+		if *p.DiscardData {
+			shouldDiscardQueryResults = true
+		}
+	}
+
+	if shouldDiscardQueryResults {
```

Comment:
Am thinking we should first `svc.ds.SaveQuery` (which will commit the query with `discard_data=true` which will cause Fleet to stop inserting results as they come in from osquery) and then delete all results.

Reply:
That's a good point; done.